OpenRA.Mods.Cnc/Traits/Minelayer.cs

```diff
@@ -99,8 +99,7 @@ void IResolveOrder.ResolveOrder(Actor self, Order order)
 			{
 				minefieldStart = cell;
 				Minefield = new[] { cell };
```

Comment:
This stomps over state needed for any already-queued PlaceMine order.

Repro case: queue up move, deploy, move, deploy, move, deploy - watch unit run around not knowing what to do.

This trait-level array will need to be removed, and the minefield cells pulled from the activity instead.

Reply:
This is actually a recent regression due to some refactoring. It worked before, but I kinda forgot why I did what I did two months ago.

This doesn't actually overwrite the state for an already queued order; It just orders the mines on the cell that the minelayer is at the moment the queued order is placed instead of where it will be when the activity comes up.

It should now be fixed.